tests/block_tools.py

```diff
@@ -2007,25 +2007,37 @@ def create_test_unfinished_block(
     )
 
 
+create_block_tools_async_count = 0
+create_block_tools_count = 0
+
+
+# xxx do the callers need different instances?
```

Comment:
yes, they do. They all need different root directories containing different `config.yaml` files. The daemon's listen port is configured in there, and the only way a test can control which listen port it uses is to write it to the config file.